src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -468,42 +402,214 @@ export class SocketBridge extends EcoBridgeChildBase {
 
     const { toAmount, serviceTime, totalGasFeesInUsd, routeId } = bestRoute
 
+    const formattedToAmount = Number(formatUnits(toAmount, result.toAsset.decimals)).toFixed(2).toString()
+
     this.store.dispatch(commonActions.setActiveRouteId(routeId))
 
     const details = {
       gas: `${totalGasFeesInUsd.toFixed(2).toString()}$`,
       estimateTime: `${(serviceTime / 60).toFixed(0).toString()} min`,
-      receiveAmount: Number(formatUnits(toAmount, toAsset.decimals)).toFixed(2).toString(),
+      receiveAmount: formattedToAmount,
       requestId: helperRequestId,
     }
 
     this.store.dispatch(this.actions.setBridgeDetails(details))
   }
 
+  public getBridgingMetadata = async () => {
+    const isBridgeSwapActive = this.store.getState().ecoBridge.ui.isBridgeSwapActive
+    const requestId = this.store.getState().ecoBridge[this.bridgeId as SocketList].lastMetadataCt
+
+    const helperRequestId = (requestId ?? 0) + 1
+
+    this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))
+
+    // reset previous data
+    this.store.dispatch(
+      this.actions.setApprovalData({
+        allowanceTarget: undefined,
+        amount: undefined,
+        chainId: undefined,
+        owner: undefined,
+        tokenAddress: undefined,
+      })
+    )
+    this.store.dispatch(this.actions.setRoutes([]))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    const { from, to } = this.store.getState().ecoBridge.ui
+
+    if (!from.chainId || !to.chainId || !this._account || !from.address || Number(from.value) === 0) return
+
+    let value = BigNumber.from(0)
+
+    //handling small amounts
+    try {
+      value = parseUnits(from.value, from.decimals)
+    } catch (e) {
+      this.store.dispatch(
+        this.actions.setBridgeDetailsStatus({
+          status: SyncState.FAILED,
+          errorMessage: 'No available routes / details',
+        })
+      )
+      return
+    }
+
+    if (isBridgeSwapActive) {
+      await this._fetchSocketRoutes({
+        fromChainId: from.chainId.toString(),
+        fromTokenAddress:
+          from.address === Currency.getNative(from.chainId).symbol ? SOCKET_NATIVE_TOKEN_ADDRESS : from.address,
+        toTokenAddress: to.address === Currency.getNative(to.chainId).symbol ? SOCKET_NATIVE_TOKEN_ADDRESS : to.address,
```

Comment:
Same here concern with symbol comparsion. On top of that, I was debugging these values:
```
{
    "from": {
        "value": "4",
        "chainId": 100,
        "address": "XDAI",
        "decimals": 18,
        "name": "xDAI",
        "symbol": "XDAI"
    },
    "to": {
        "value": "",
        "chainId": 42161,
        "address": "",
        "decimals": 0
    },
    "value": {
        "type": "BigNumber",
        "hex": "0x3782dace9d900000"
    }
}
```

`"address": "XDAI"` is a bit concerning. What do you think?

Reply:
same https://github.com/levelkdev/swapr-dapp/pull/1253#discussion_r953821406
I think we can improve it later because every bridge checks native currency like this.

Reply:
Ok, let's jump on a call and create follow-up issues :)